zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -361,13 +361,22 @@ func (ob *Observer) GetInboundVoteFromBtcEvent(event *BTCInboundEvent) *crosscha
 	}
 	amountInt := big.NewInt(amountSats)
 
-	// create inbound vote message contract V1 for legacy memo
+	// create inbound vote message contract V1 for legacy memo or standard memo
+	var msg *crosschaintypes.MsgVoteInbound
 	if event.MemoStd == nil {
-		return ob.NewInboundVoteFromLegacyMemo(event, amountInt)
+		msg = ob.NewInboundVoteFromLegacyMemo(event, amountInt)
+	} else {
+		msg = ob.NewInboundVoteFromStdMemo(event, amountInt)
 	}
 
-	// create inbound vote message for standard memo
-	return ob.NewInboundVoteFromStdMemo(event, amountInt)
+	// make sure the message is valid before posting to zetacore
+	err = msg.ValidateBasic()
+	if err != nil {
+		ob.Logger().Inbound.Error().Err(err).Fields(lf).Msg("invalid inbound vote message")
+		return nil
```

Comment:
Consider returning `(*crosschaintypes.MsgVoteInbound, error)` or (msg *crosschaintypes.MsgVoteInbound, skipped bool, err error). 

- Currently, it's confusing the func returns a pointer to a structure that might be nil due to an error 
- What if validateBasic returns an error due to a mistake in tx parsing in zetaclient (assuming raw btc tx is valid)?

Reply:
That's a good point. Any error will result in a `nil` Msg, which means that we should SKIP the event. We trust the `validateBasic` function because it is the same method used by `checkTx` in the `zetacored` binary.

But I added some comments to the function for better readability: https://github.com/zeta-chain/node/pull/3184/commits/b00341083ef4a443d844bac59a0b4c178797d25f